certbot/certbot/_internal/main.py

```diff
@@ -1537,6 +1537,12 @@ def certonly(config: configuration.NamespaceConfig, plugins: plugins_disco.Plugi
 
     lineage = _get_and_save_cert(le_client, config, domains, certname, lineage)
 
+    # If a new cert was issued and we were passed an installer, we can safely
+    # run `installer.restart()` to load the newly issued certificate
+    if lineage and installer and not config.dry_run:
+        display_util.notify(f"Reloading {config.installer} server after certificate issuance")
+        installer.restart()
```

Comment:
Calling `restart` on an installer can raise an error. Since we already saved the certificate at this point, we probably want the rest of the function to be crash-proof.

In the implementation of `run`, we track whether the installer raised an error and later print a message, telling the user that they need to run `certbot install` after fixing the issue.

We might want to do something similar here, I'm not sure yet.

Reply:
Thinking about this some more, I think it would be consistent to pass the `installer.restart()` error onto the reporting function, like how `certbot run` does.

We will probably need different wording for that case, since asking the user to run `certbot install` doesn't make sense for `certonly`.